File: jvm/src/main/scala/org/eigengo/sogx/config/DispatcherServletInitializer.scala
package org.eigengo.sogx.config

import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer
import javax.servlet.{ServletContext, ServletRegistration}

class DispatcherServletInitializer extends AbstractAnnotationConfigDispatcherServletInitializer {

  protected def getRootConfigClasses: Array[Class[_]] = {
    Array[Class[_]](classOf[Webapp])
  }

  protected def getServletConfigClasses: Array[Class[_]] = {
    Array[Class[_]](classOf[Webapp])
  }

  protected def getServletMappings: Array[String] = {
    Array[String]("/")
  }

  protected override def customizeRegistration(registration: ServletRegistration.Dynamic) {
    registration.setInitParameter("dispatchOptionsRequest", "true")
  }

  override def onStartup(servletContext: ServletContext) {
    servletContext.setInitParameter("spring.profiles.active", "simple-broker")
    super.onStartup(servletContext)
  }
}

File: jvm/web/src/main/java/org/eigengo/sogx/web/DispatcherServletInitializer.java
package org.eigengo.sogx.web;

import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.ServletRegistration;
import javax.servlet.ServletRegistration.Dynamic;

import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;

public class DispatcherServletInitializer extends AbstractAnnotationConfigDispatcherServletInitializer {

	@Override
	protected Class<?>[] getRootConfigClasses() {
		return new Class<?>[] { WebConfig.class };
	}

	@Override
	protected Class<?>[] getServletConfigClasses() {
		return new Class<?>[] { WebConfig.class };
	}

	@Override
	protected String[] getServletMappings() {
		return new String[] { "/" };
	}

	@Override
	protected void customizeRegistration(ServletRegistration.Dynamic registration) {
		registration.setInitParameter("dispatchOptionsRequest", "true");
	}

	@Override
	public void onStartup(ServletContext servletContext) throws ServletException {
		servletContext.setInitParameter("spring.profiles.active", "simple-broker");
//		servletContext.setInitParameter("spring.profiles.active", "stomp-broker-relay");
		super.onStartup(servletContext);
	}

}
